lib/elixir/lib/process.ex

```diff
@@ -839,13 +839,17 @@ defmodule Process do
     nilify(:erlang.process_info(pid))
   end
 
+  @type process_info_item :: atom | {:dictionary, term}
+  @type process_info_result_item :: {process_info_item, term}
+
   @doc """
   Returns information about the process identified by `pid`,
   or returns `nil` if the process is not alive.
 
   See `:erlang.process_info/2` for more information.
   """
-  @spec info(pid, atom | [atom]) :: {atom, term} | [{atom, term}] | nil
+  @spec info(pid, process_info_item | [process_info_item]) ::
+          process_info_result_item | [process_info_result_item] | nil
```

Comment:
```suggestion
  @spec info(pid, process_info_item) :: process_info_result_item | nil
  @spec info(pid, [process_info_item]) :: [process_info_result_item] | nil
```

I think this could work as well and be better for documentation (althoguh maybe Dialyzer would unify these types internally?)

Reply:
Pleasantly surprised to report that this improves the spec for Dialyzer as well:
for `{:should_warn, _} = Process.info(self(), [:some_atom])`:
```
The pattern
          {'should_warn', _} can never match the type
          'nil' | [{atom() | {'dictionary', _}, _}]
```

and for `[{:should_warn, _}] = Process.info(self(), :some_atom)`
```
lib/task/supervised.ex:238: The pattern
          [{'should_warn', _}] can never match the type
          'nil' | {atom() | {'dictionary', _}, _}
```